pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -382,14 +382,25 @@ def _resolve_sampler(self, dataloader: DataLoader, shuffle: bool, mode: Optional
                     " distributed training. Either remove the sampler from your DataLoader or set"
                     " `replace_sampler_ddp=False` if you want to use your custom sampler."
                 )
-            return self._get_distributed_sampler(
+            sampler = self._get_distributed_sampler(
                 dataloader,
                 shuffle,
                 mode=mode,
                 overfit_batches=self.trainer.overfit_batches,
                 **self.trainer.distributed_sampler_kwargs,
             )
 
+            trainer_fn = self.trainer.state.fn
+            if isinstance(sampler, DistributedSampler) and trainer_fn in (TrainerFn.VALIDATING, TrainerFn.TESTING):
+                rank_zero_warn(
```

Comment:
Since this information is already printed here, do you still think it's worth adding the same to the docs?

Reply:
yeah, I think it is quite informative and good to have inside docs too for more visibility. If someone is just getting started and find this in docs, they might configure it accordingly then.

Reply:
If you prefer too have same message in both, then I'd suggest you add a comment here so we don't forget